roles/addons/nic_nmcli/tasks/main.yml

```diff
@@ -65,7 +65,10 @@
     # {% endif %}
     # Allows to provide a simple ip4, or an ip4/prefix
     # Also support old prefix format (to be replaced by prefix4)
-    gw4: "{{ item.gw4 | default(networks[item.network].gateway4) | default(networks[item.network].gateway) | default(omit) }}"
+    gw4: "{{ (item.never_default4 | default(networks[item.network].never_default4) is not defined) | ternary( item.gw4 | default(networks[item.network].gateway4) | default(networks[item.network].gateway) | default(omit), omit) }}"
+    routes4: "{{ (item.routes4 is defined or networks[item.network]['routes4'] is defined) |ternary( item.routes4 | default(networks[item.network]['routes4'] | default([]) ), omit) }}"
```

Comment:
The `| default([])` seems unnecessary. When `(item.routes4 is defined or networks[item.network]['routes4'] is defined)` is true, one of both is defined, so `item.routes4 | default(networks[item.network]['routes4'])` should be sufficient. If `item.routes4` is not defined, we know that `networks[item.network]['routes4']` is. Or the first condition is false, then omit.
I wonder if you could achieve the same using a logic similar to route_metric4 below.
Am I missing something?

Reply:
😉 Good question: in fact, for an unknown reason, Ansible try to grab all values inside the ternary, the ones from true and the ones from false. So in ternary, you always need to provide default[] if you are not sure a value can exist.